### PR TITLE
[WC-1631] Association filter feedback

### PR DESCRIPTION
**Pull request by iobuhov**
### Description

Small adjustments in Drop-down filter and e2e tests

### Pull request type

-   [ ] No code changes (changes to documentation, CI, metadata, etc)
-   [ ] Dependency changes (any modification to dependencies in `package.json`)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [x] Refactoring
-   [x] Test related change (New E2E test, test automation, etc.)

### CODE REVIEW

File: packages/pluggableWidgets/datagrid-web/package.json [hunk 1]
```diff
@@ -19,12 +19,12 @@
...
   },
   "marketplace": {
-    "minimumMXVersion": "9.6.0",
+    "minimumMXVersion": "9.17.0",
     "appName": "Data Grid 2"
   },
   "testProject": {
     "githubUrl": "https://github.com/mendix/testProjects",
-    "branchName": "datagrid-web"
+    "branchName": "datagrid-web-next"
```
Comment (r0b1n) [inline packages/pluggableWidgets/datagrid-web/package.json:27]: Didn't we want to keep this pointing at the current name but change legacy versions to point to a new legacy branch?

File: packages/pluggableWidgets/datagrid-web/src/Datagrid.editorConfig.ts [hunk 2]
```diff
@@ -298,62 +294,84 @@ export function check(values: DatagridPreviewProps): Problem[] {
...
-            errors.push(attributeError);
+        if (column.sortable && !column.attribute) {
+            return {
+                property: columnPropPath("attribute", index),
+                message: `An attribute is required when column sorting is enabled. Select the 'Attribute' property for column (${column.header}) or disable sorting in column settings`
+            };
+        }
+    };
+
+    const columnChecks = [checkAssociationSettings, checkFilteringSettings, checkDisplaySettings, checkSortingSettings];
```
Comment (r0b1n) [inline packages/pluggableWidgets/datagrid-web/src/Datagrid.editorConfig.ts:367]: Could we move those checking functions outside of the `check` function. That way they are not re-created every time `check` is called.

File: packages/pluggableWidgets/datagrid-web/package.json (hunk 1 above)
Comment (iobuhov) [reply packages/pluggableWidgets/datagrid-web/package.json:27]: I'll switch to normal name as soon as this PR is merged. The idea is to not break other PRs. Once we merge it to main, we can rebase other PRs + I'll open new PR to switch to default name.

File: packages/pluggableWidgets/datagrid-web/src/Datagrid.editorConfig.ts (hunk 2 above)
Comment (iobuhov) [reply packages/pluggableWidgets/datagrid-web/src/Datagrid.editorConfig.ts:367]: Done 🪄

**Review by r0b1n — APPROVED** [review]

**Review by leonardomendix — APPROVED** [review]

**Pull request merged**